### ISSUE TITLE
Clang treat uint32_t on MacOS as int

### ISSUE THREAD

**Issue by QuantumDeveloper** [mono/CppSharp#1285]
Hi to all. 
I know that this repo is not about clang, but I dont know another place. On SO noone wants to answer clang questions so I will ask here if you dont mind.

I used clang on MacOS to generate bindings to Vulkan and found that clang treats `uint32_t` and `uint64_t ` as int and long instead `uint `and `ulong` as it does on WIndows. 

So, I want to ask is this by design? Because typedef is pointing to `typedef unsigned int uint32_t;`

Does someone faced with this and know how to fix it?
P.S. I am using clang 11.0 from xcode

**Comment by tritao** [comment]
This is using standalone Clang API, and not through CppSharp right?

Did you set the correct triple?

**Comment by QuantumDeveloper** [comment]
Yeah, correct. This is using libclang.dylib
What do you mean by triple?

**Comment by QuantumDeveloper** [comment]
@tritao Is there any suggestions regarding this issue?

**Comment by ddobrev** [comment]
@QuantumDeveloper not by me at present but I can tell you that I've asked in the IRC channel of LLVM many times and with some patience there always is someone who can help. You can try there.

**Comment by QuantumDeveloper** [comment]
@ddobrev How I can get there? Can you point me please?

**Comment by ddobrev** [comment]
@QuantumDeveloper you can find it at https://llvm.org/ -  irc.oftc.net #llvm. Then you use any IRC client you like, or just the browser if you so prefer.

**Comment by tritao** [comment]
.Is there any reason that you can't use the parsing and AST support provided by CppSharp?

Any limitations that might not fit your use case?

We use the Clang C++ API directly, we don't have much experience with using libclang / C API, in fact we found a couple limitations that forced us to migrate to the C++ API years ago.

**Comment by QuantumDeveloper** [comment]
@tritao  Unfortunately CppSharp not fully fit my requirement (it very cool lib, but I found some limitations) For current moment I wrote my own analogue of CppSharp (maybe with less functionality) but it generates the interop code, which exactly match my needs and can be configured at most points (because C code sometimes in unpredictable from the parsing perspective and a lot of method should be configured manually)
Thats why I decided to write my own lib using libclang.

Anyway, thanks, I will ask clang developers at discord channel. Didnt see they have it.
